### ISSUE TITLE
Android 8.0 issue

### ISSUE THREAD

**Issue by sahuadarsh0** [Mauker1/MaterialSearchView#171]
```
 java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter previouslyFocusedRect
        at br.com.mauker.materialsearchview.MaterialSearchView.requestFocus(Unknown Source:2)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3163)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3160)
        at android.view.ViewGroup.onRequestFocusInDescendants(ViewGroup.java:3204)
        at android.view.ViewGroup.requestFocus(ViewGroup.java:3163)
        at android.view.View.requestFocus(View.java:10773)
        at android.view.View.requestFocus(View.java:10715)
        at android.view.ViewRootImpl.focusableViewAvailable(ViewRootImpl.java:3430)
        at android.view.View.setFlags(View.java:13281)
        at android.view.View.setVisibility(View.java:9382)
        at android.app.Activity.makeVisible(Activity.java:5412)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3785)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2898)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
```

**Comment by Mauker1** [comment]
Can you give us more details on what were you trying to do before this error showed up? Which method did you call?

**Comment by sahuadarsh0** [comment]
the error is because of the view I guess in android 8.0 API 26-28 this error is showing , after making visibility gone of the material search view the error is gone

**Comment by sahuadarsh0** [comment]
This issue is not available in version 1.2.3

**Comment by hardysim** [comment]
I think I have the same problem. It occurs when submitting the search-query via the enter-key or the submit-button of the virtual keyboard.

In this case, `ViewGroup.requestFocus()` is called which in turn calls `MaterialSearchView.requestFocus()` with `previouslyFocusedRect` set to `null`. And that's the error: the kotlin-signature expects a non-null `Rect` (must be `Rect?`).

This is not happening in the demo-app because nothing is done when submitting. In my case, I navigate to a different fragment and this might trigger `requestFocus()`.

This also happens with the version *2.0.0-alpha03*.
